### PR TITLE
Make strings `const` in the `init_builtin_traits` function

### PR DESCRIPTION
**Pull request by yuk1ty**
Replaced hard-coded strings with constants.

#266 

@mtshiba

### CODE REVIEW

File: compiler/erg_compiler/context/initialize/mod.rs [hunk 1]
```diff
@@ -44,7 +44,59 @@ use Visibility::*;
 
 const NUM: &str = "Num";
 
-const NAME: &str = "__name__";
+const UNPACK: &str = "Unpack";
+const INHERITABLE_TYPE: &str = "InheritableType";
+const NAMED: &str = "Named";
+const MUTABLE: &str = "Mutable";
+const SELF: &str = "Self";
+const IMMUTIZABLE: &str = "Immutizable";
```
Comment (yuk1ty) [inline compiler/erg_compiler/context/initialize/mod.rs:52]: nits: should be "Immutable"?

File: compiler/erg_compiler/context/initialize/mod.rs [hunk 2]
```diff
@@ -44,7 +44,59 @@ use Visibility::*;
...
+const IMMUT_TYPE: &str = "ImmutType";
+const PROC_UPDATE: &str = "update!";
+const MUTIZABLE: &str = "Mutizable";
```
Comment (yuk1ty) [inline compiler/erg_compiler/context/initialize/mod.rs:55]: nits: should be "Mutable"?

File: compiler/erg_compiler/context/initialize/mod.rs (hunk 1 above)
Comment (mtshiba) [reply compiler/erg_compiler/context/initialize/mod.rs:52]: This is not a mistake. An `Immutizable` object is an object that can be converted to an `Immutable` object (e.g. `Int!` -> `Int`). It is not itself `Immutable`. However, `Immutizable` is not a common adjective, so I'm trying to think of a better name.

File: compiler/erg_compiler/context/initialize/mod.rs (hunk 2 above)
Comment (mtshiba) [reply compiler/erg_compiler/context/initialize/mod.rs:55]: This is also a separate trait from `Mutable` for the same reason as `Immutizable`.

File: compiler/erg_compiler/context/initialize/mod.rs (hunk 1 above)
Comment (yuk1ty) [reply compiler/erg_compiler/context/initialize/mod.rs:52]: Understood. Thank you for the elaboration!

**Review by mtshiba — APPROVED** [review]

**Comment by mtshiba** [comment]
Thanks!

**Pull request merged**